src/pages/ValidateLoginNewWorkspacePage.js

```diff
@@ -39,7 +45,11 @@ class ValidateLoginNewWorkspacePage extends Component {
             // by calling dismissModal(), the /v/... route is removed from history so the user will get taken to `/`
             // if they cancel out of the new workspace modal.
             Navigation.dismissModal();
-            Navigation.navigate(ROUTES.WORKSPACE_NEW);
+            if (_.isEmpty(this.props.betas)) {
+                setRedirectToWorkspaceNewAfterSignIn(true);
```

Comment:
@chiragsalian I'm not quite understanding this, so can you explain it to me? If this logic runs, there is already an authToken, so why is the user being redirected to the new workspace page after sign in, when they are already signed in?

Reply:
Maybe the method name is not the best. Atm i just copied pasted the param i.e., `redirectToWorkspaceNewAfterSignIn`. The functionality of this is more like `redirectAfterSignInAndHaveNecessaryParams`. So yeah even though we have the authToken we don't have the betas loaded yet. 
So if we go to `Navigation.navigate(ROUTES.WORKSPACE_NEW)` without the betas it will fail and not show the component, which is the issue.

The betas take a little longer to load after being signed in which is why we save the redirect in onyx and then check if we have everything [here](https://github.com/Expensify/App/pull/4205/files#diff-2a28868769d50b974461521d1d323f02f49c907ea488c6e79a7b058a8041017eR141-R142) before we can navigate to the component. Does that make sense?

As for the method name. Atm i just renamed it exactly to the variable. I thought once I continue with the QR code redirect project of mine I'll discuss the method name closely with you to make it more generic.

Reply:
OK, I understand about needing to wait for betas to load, but I'm still confused why `setRedirectToWorkspaceNewAfterSignIn` is necessary. It wasn't in the original code, so it's being added for some reason, but I can't tell what that reason is.

Reply:
> It wasn't in the original code, so it's being added for some reason, but I can't tell what that reason is.

Do you mean you can't tell easily tell what the reason is in the code and hence want me to add comments or you can't see the reason in general. Cause the reason its added is to fix the [GH issue](https://github.com/Expensify/Expensify/issues/171345). Since its possible to sign in faster than getting betas, if we redirect to `Navigation.navigate(ROUTES.WORKSPACE_NEW)` too early then the new workspace screen won't show up. You can easily reproduce the same on dev too with my test steps (Test step - `Testing pricing redirect` while being signed out hits this flow). Let me know if that makes sense.

Reply:
Discussed 1:1 and landing on a resolution to be made in my follow of QR scanning project work.